code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,329 @@
+	desc = "A computer console equipped with camera screen and controls for a planetside deployed construction drone. Materials or equipment vouchers can be added simply by inserting them into the computer."
+	icon_state = "syndishuttle"
+	req_access = list(ACCESS_MARINE_REMOTEBUILD)
+	networks = FALSE
+	off_action = new/datum/action/innate/camera_off/remote_fob
+	jump_action = null
+	var/drone_creation_allowed = TRUE
+	var/obj/docking_port/stationary/marine_dropship/spawn_spot
+	var/lz
+	var/datum/action/innate/remote_fob/metal_cade/metal_cade = new
+	var/max_metal = 50 //mostly to prevent jokers collecting all the metal and dumping it in
+	var/metal_remaining = 0
+	var/datum/action/innate/remote_fob/plast_cade/plast_cade = new
+	var/max_plasteel = 50
+	var/plasteel_remaining = 0
+	var/datum/action/innate/remote_fob/toggle_wiring/toggle_wiring = new //whether or not new barricades will be wired
+	var/do_wiring = FALSE
+	var/datum/action/innate/remote_fob/sentry/sentry = new
+	var/sentry_remaining = 0
+
+/obj/machinery/computer/camera_advanced/remote_fob/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT, .proc/disable_drone_creation)
+
+/obj/machinery/computer/camera_advanced/remote_fob/proc/disable_drone_creation()
+	drone_creation_allowed = FALSE
+	UnregisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT)
+
+/obj/machinery/computer/camera_advanced/remote_fob/examine(mob/user)
+	. = ..()
+	var/list/details = list()
+	details +=("It has [metal_remaining] sheets of metal remaining.</br>")
+	details +=("It has [plasteel_remaining] sheets of plasteel remaining.</br>")
+	details +=("It has [sentry_remaining] sentries ready for placement.</br>")
+	to_chat(user, "[details.Join(" ")] ")
+
+/obj/machinery/computer/camera_advanced/remote_fob/give_eye_control(mob/user)
+	. = ..()
+	user.lighting_alpha = 120
+	eyeobj.name = "Remote Construction Drone"
+	if(eyeobj.eye_initialized)
+		eyeobj.setLoc(get_turf(spawn_spot))
+
+
+/obj/machinery/computer/camera_advanced/remote_fob/attack_hand(mob/living/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='warning'>Access Denied!</span>")
+		return
+	if(!drone_creation_allowed)
+		to_chat(user, "<span class='notice'>Communication with the drone impossible due to fuel-residue in deployment zone atmosphere.</span>")
+		return
+	spawn_spot = FALSE
+	switch(tgalert(user, "Summon Drone in:", "FOB Construction Drone Control", "LZ1","LZ2"))
+		if("LZ1")
+			spawn_spot = locate(/obj/docking_port/stationary/marine_dropship/lz1) in SSshuttle.stationary	
+		if("LZ2")
+			spawn_spot = locate(/obj/docking_port/stationary/marine_dropship/lz2) in SSshuttle.stationary
+		else
+			return
+	return ..()
+	
+
+/obj/machinery/computer/camera_advanced/remote_fob/CreateEye()
+	if(spawn_spot)
+		eyeobj = new /mob/camera/aiEye/remote/fobdrone(get_turf(spawn_spot))
+		eyeobj.origin = src
+	return
+
+/obj/machinery/computer/camera_advanced/remote_fob/attackby(obj/item/stack/W, mob/user, params)
```

Comment:
```suggestion
/obj/machinery/computer/camera_advanced/remote_fob/attackby(obj/item/attacking_item, mob/user, params)
```
The base of `attackby` is with items. Do not worry about having to typecast it if you are sure that right now you'll only deal with stacks, but don't just retype it in the proc call, because else you are passing the wrong information to further coders that you can only get that kind of type of input on this proc.

Reply:
this is done to access stack-specific stuff, notably in this case the amount var, if there's a better way to do that i'm all ears

Reply:
Something like this:
```
/obj/machinery/computer/camera_advanced/remote_fob/attackby(obj/item/attacking_item, mob/user, params)
	if(istype(attacking_item, /obj/item/stack/voucher/sentry))
		var/obj/item/stack/attacking_stack = attacking_item
		var/useamount = attacking_stack .amount
```